src/api/app/controllers/webui/package_controller.rb

```diff
@@ -178,7 +180,10 @@ def binary
       return
     end
 
-    @durl = download_url_for_file_in_repo(@project, @package_name, @repository, @arch.name, @filename)
+    url_generator = ::PackageControllerService::URLGenerator.new(project: @project, package: @package,
```

Comment:
You need to use @package_name here, as it can be different from `@package` and it will point to the wrong url!!!!!

Reply:
@DavidKang can you please review it again?

Reply:
exactly, this was introduced to handle multibuilds. Because of that it can not be removed :wink: